DMTN-148.tex

```diff
@@ -50,9 +50,237 @@
+allows the calibration to be regenerated by any user, making the
+construction process much more transparent than it currently is.  A
+YAML file containing the dataId parameters for all inputs would
+satisfy this.  Appending additional construction statistics (for
+example, weight factors applied to each input during the combination
+stage) would be helpful in the validation process.
+
+Newly generated calibration products will belong to a Gen3 collection,
+and that collection will be used to manage the calibration through the
+validation and certification process.  This collection may only
+contain one entry for a given OIDF set.
+
+\subsection{Validation}
+
+Calibration products are not currently validated, largely due to the
+opaque construction process.  Some set of proper validation tasks must
+be made, applying the newly constructed product to a set of test
+exposures prior to that calibration product's certification.  Applying
+these validation tasks continuously as new data is taken will inform
+the date ranges for which the product is certified.
+
+\subsection{Certification}
+
+Once a calibration product has been confirmed through validation to
+correct the instrument effect it is designed for, it will be certified
+with a date range denoting when the calibration is valid, and assigned
+to a particular calibration collection (which may contain calibrations
+the same OIDF values, as the validity range adds another dimension).
+For FITS image calibrations, this is equivalent to a database
+operation; no persisted data need change.  This is also acceptable for
+text calibrations, although conversion to FITS (using the
+CalibrationBaseClass methods) may be desired for IO speed. In
+addition, as the text calibrations are human readable and easier to
+manage, they should be exported to the appropriate \verb|obs_CAMERA_data|
+package.  This provides a common versioned repository for the state of
+the camera calibration that can be exported and used independently of
+the Gen3 repository where they were constructed.  These exports should
+also include the configuration and input data records.
+
+\subsection{Modifying Calibrations}
+
+Most calibration products need user intervention only to launch the
+construction and to validate and certify the result.  However, it is
+assumed that some of the products used (such as \verb|CAMERA| and \verb|QE_CURVE|
+definitions) will not have a PipelineTask construction process.  The
+version of these products stored in the \verb|obs_CAMERA_data| (or \verb|obs_CAMERA|
+for the case of the \verb|CAMERA| calibration) repository will be ingested
+into the Butler repository as needed.  These products should still
+have as much configuration and provenance data as is possible, but
+should also note any manual edits with a name and date.  This
+provenance method should also be used for calibrations that need to be
+augmented in non-algorithmic ways, such as \verb|DEFECTS|.
```

Comment:
I do wonder whether the command line tool that does the ingestion of these calibrations directly from the git repo to the butler should include the obs_CAMERA_data SHA1 somewhere -- maybe in the default collection name?

Reply:
My assumption on this is that a new calibration that has been modified will either have a new calibration date range (amplifier XYZ went bad on this date, and is now masked), or will supplant a previously ingested calibration (calibration XYZ has always been bad, but it's now fixed as of f5624ed).  That said, I think anything that goes into a git repo must come out with the SHA1 attached in the metadata, and using that as the default collection name with the OBSTYPE makes sense to me.